### PR TITLE
The case of identifier names of db objects is preserved now

### PR DESCRIPTION
**Pull request by h-thurow**
@brettwooldridge 

The case of identifier names of db objects is preserved now: As discussed. Identifiers can be delimited to support non standard conform identifier names.

Support for delimited fields with `@GeneratedValue` annotation: Setting the id (OrmWriter#fillGeneratedId()) did not work when field name was delimited.

Optionally exclude columns from being updated when calling OrmElf.updateObject(...): This is to exclude a field per statement instead excluding it always with "updatable = false". It mimics OrmElf#resultSetToObject(java.sql.ResultSet, T, java.util.Set<java.lang.String>). The statement will not be cached.

### CODE REVIEW

**Comment by brettwooldridge** [comment]
Super awesome work!  I’ll review it.

File: src/main/java/com/zaxxer/sansorm/internal/Introspected.java [hunk 1]
```diff
@@ -318,6 +320,7 @@ public boolean hasGeneratedId()
       List<String> columns = new LinkedList<>();
       if (hasGeneratedId()) {
          columns.addAll(Arrays.asList(columnsSansIds));
+         removeColumns(columns, entry -> Introspected.this.isInsertableColumn(entry.getKey()));
```
[the remark this answers is not in the document]
Comment (h-thurow) [reply src/main/java/com/zaxxer/sansorm/internal/Introspected.java]: removeColumns() removes from its first argument "columns" all those columns "check" does not return true for. Should I rename it to retainColumns() anyway?

[the remark this answers is not in the document]
Comment (h-thurow) [reply src/main/java/com/zaxxer/sansorm/internal/Introspected.java]: isInsertableColumn() is called from two places:
```
Introspected.getDelimitedInsertableColumns(List<String>)
Introspected.getInsertableColumns()
```
I will try to precalculate it too.

[the remark this answers is not in the document]
Comment (brettwooldridge) [reply src/main/java/com/zaxxer/sansorm/internal/Introspected.java]: When I read that line...
```java
removeColumns(columns, entry -> Introspected.this.isInsertableColumn(entry.getKey()));
```
It *reads* to me like, "Remove the insertable columns from ``columns``".  But I think what it does is remove all *non*-insertable columns, correct?

[the remark this answers is not in the document]
Comment (h-thurow) [reply src/main/java/com/zaxxer/sansorm/internal/Introspected.java]: DONE removeColumns() renamed to retainColumns()

**Comment by brettwooldridge** [comment]
@h-thurow Nice work, Thanks!

File: src/main/java/com/zaxxer/sansorm/internal/OrmBase.java [hunk 2]
```diff
@@ -151,4 +151,16 @@ protected static Object mapSqlType(Object object, int sqlType)
...
+   /**
+    * Case insensitive comparison.
+    */
+   protected static boolean isIgnoredColumn(Set<String> ignoredColumns, String columnName) {
+      for (String ignoredColumn : ignoredColumns) {
+         if (columnName.compareToIgnoreCase(ignoredColumn) == 0) {
+            return true;
+         }
+      }
+      return false;
```
[the remark this answers is not in the document]
Comment (h-thurow) [reply src/main/java/com/zaxxer/sansorm/internal/OrmBase.java]: DONE Stream API

**Pull request merged**